sklearnex/cluster/k_means.py

```diff
@@ -1,17 +1,370 @@
+                self._check_params_vs_input(X)
+            else:
+                self._check_params(X)
+
+            self._n_features_out = self.n_clusters
+            self._n_threads = _openmp_effective_n_threads()
+
+            self._initialize_onedal_estimator()
+            self._onedal_estimator.fit(X, queue=queue)
+
+            self._save_attributes()
+
+        def _onedal_predict_supported(self, method_name, X, sample_weight=None):
+            assert method_name == "predict"
+
+            class_name = self.__class__.__name__
+            is_data_supported = (
+                _is_csr(X) and daal_check_version((2024, "P", 600))
+            ) or not issparse(X)
+            patching_status = PatchingConditionsChain(
+                f"sklearn.cluster.{class_name}.predict"
+            )
+
+            supported_algs = ["auto", "full", "lloyd", "elkan"]
+
+            patching_status.and_conditions(
+                [
+                    (
+                        self.algorithm in supported_algs,
+                        "Only lloyd algorithm is supported, elkan is computed using lloyd.",
+                    ),
+                    (
+                        is_data_supported,
+                        "Supported data formats: Dense, CSR (oneDAL version >= 2024.6.0).",
+                    ),
+                    (
+                        hasattr(self, "_onedal_estimator"),
+                        "oneDAL model was not fit.",
+                    ),
+                ]
+            )
+
+            return patching_status
+
+        if sklearn_check_version("1.5"):
+
+            @wrap_output_data
+            def predict(self, X):
+                self._check_feature_names(X, reset=False)
+                self._validate_params()
+                return dispatch(
+                    self,
+                    "predict",
+                    {
+                        "onedal": self.__class__._onedal_predict,
+                        "sklearn": sklearn_KMeans.predict,
+                    },
+                    X,
+                )
+
+        else:
+
+            @wrap_output_data
+            def predict(
+                self,
+                X,
+                sample_weight="deprecated" if sklearn_check_version("1.3") else None,
+            ):
+                if sklearn_check_version("1.0"):
+                    self._check_feature_names(X, reset=False)
+                if sklearn_check_version("1.2"):
+                    self._validate_params()
+                return dispatch(
+                    self,
+                    "predict",
+                    {
+                        "onedal": self.__class__._onedal_predict,
+                        "sklearn": sklearn_KMeans.predict,
+                    },
+                    X,
+                    sample_weight,
+                )
+
+        def _onedal_predict(self, X, sample_weight=None, queue=None):
+            X = self._validate_data(
```

Comment:
Add a sklearn_check_version("1.0") here and in the else use a check_X_y

Reply:
We are not probably supporting sklearn<1.0 any more, do you think we still need this?